src/Map/ProspectorOverlayLayer.cs

```diff
@@ -408,7 +406,7 @@ private void OnChatMessage(int groupId, string message, EnumChatType chattype, s
             var newProspectInfo = new ProspectInfo(posX, posZ, message);
             _prospectInfos.RemoveAll(m => m.X == posX && m.Z == posZ);
             _prospectInfos.Add(newProspectInfo);
-            _clientApi.SaveDataFile(Filename, _prospectInfos); //TODO saving the data every time is quite overkill. Should use a more efficient approach.
+            _prospectInfos.HasChanged = true;
```

Comment:
What happens with the data when the client crashes (like killing it with task-manager), does the player lose all progress since the last rejoin?

Reply:
Maybe we can save the data after every x prospecting actions?

Reply:
i like the general idea but i think there could potentially be a long time between reaching the thresholds which would result in lost data (in contrast to now where we call save every prospecting). i mean we could also register a RegisterGameTickListener at startup and call save/reset the dirty flag every once in a while (like checking once a minute)

Reply:
Fair point. I think there are two (potentially) opposing goals here:
1. Do not lose prospecting data from current session.
2. Reduce write wear on disk. I guess this was also the concern of the TODO comment.

I think 1. is indeed important but I have no idea how often VS crashes. I like your idea of periodically storing the data, though I would choose a larger interval (10-20 mins) betweens saves, if any new data was added.